src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -409,6 +409,7 @@ public void Save(Stream stream)
             ValidateDependencies(Version, DependencyGroups);
             ValidateReferenceAssemblies(Files, PackageAssemblyReferences);
             ValidateFrameworkAssemblies(FrameworkReferences, FrameworkReferenceGroups);
+            ValidateFilesUnique(Files);
```

Comment:
Shouldn't this be validated earlier? If there are dupes, then we'd validate reference assemblies multiple times and then later fail the Unique file check.

Reply:
how about now?